packages/cli-repl/src/cli-repl.ts

```diff
@@ -344,6 +353,40 @@ class CliRepl {
     }
   }
 
+  async loadGlobalConfigFile(): Promise<Partial<CliUserConfig>> {
+    let fileContents = '';
+    let filename = '';
+    for (filename of this.globalConfigPaths) {
+      try {
+        fileContents = await fs.readFile(filename, 'utf8');
+        break;
+      } catch { /* ignore error */ }
+    }
+    this.bus.emit('mongosh:globalconfig-load', { filename, found: fileContents.length > 0 });
+    try {
+      let config: CliUserConfig;
+      if (fileContents.trim().startsWith('{')) {
+        config = bson.EJSON.parse(fileContents) as any;
+      } else {
+        config = (yaml.load(fileContents) as any)?.mongosh ?? {};
+      }
+      for (const [key, value] of Object.entries(config) as [keyof CliUserConfig, any][]) {
+        const validationResult = await CliUserConfigValidator.validate(key, value);
+        if (validationResult) {
+          const msg = `Warning: Ignoring config option "${key}" from ${filename}: ${validationResult}\n`;
+          this.output.write(this.clr(msg, ['bold', 'yellow']));
```

Comment:
A bit unrelated, so maybe we do nothing for now - We have these `bold', 'yellow'` warning settings in a few places in mongosh, would it make sense to have them shared as a value somewhere? Maybe in `format-output.ts`?

Reply:
Yeah, I really love the idea. We already have centralized color handling in `clr.ts`, I’ve put something there (ac52225702e1d127f212941895cf652ea90273f8) :)